### PR TITLE
feat(web): implement asset in content page

### PR DESCRIPTION
**Pull request by nourbalaha**
# Overview
In this PR I implemented asset in content page.

### CODE REVIEW

File: web/src/components/organisms/Project/Content/ContentDetails/index.tsx [hunk 1]
```diff
@@ -15,10 +18,35 @@ const ContentDetails: React.FC = () => {
...
+    handleSearchTerm,
+    handleAssetsReload,
+    loading,
+    createAssets,
+    fileList,
+    setFileList,
+    setUploading,
+    setUploadModalVisibility,
+    uploading,
+    uploadModalVisibility,
```
Comment (KaWaite) [inline web/src/components/organisms/Project/Content/ContentDetails/index.tsx:33]: I already asked @issmail-basel to do this too. Can prop order/hook return order be variables followed by functions? And of course, variables related to one another should be together and functions related to one another should be together.

File: web/src/components/molecules/Common/Form/AssetItem/index.tsx [hunk 2]
```diff
@@ -0,0 +1,125 @@
...
+    setFileList([]);
+  };
+
+  const handleUpload = () => {
+    setUploading(true);
+    createAssets(fileList).finally(() => {
+      hideUploadModal();
+      // TODO: link the uploaded asset with content after uploading is done
+    });
+  };
```
Comment (KaWaite) [inline web/src/components/molecules/Common/Form/AssetItem/index.tsx:72]: useCallbacks

Comment (KaWaite) [reply web/src/components/molecules/Common/Form/AssetItem/index.tsx:72]: Also can you use async/await. In most of our codebases we use async/await instead of `someFunction().finally()`

File: web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx [hunk 3]
```diff
@@ -0,0 +1,150 @@
...
+  fileList: UploadFile<File>[];
+  uploading: boolean;
+  uploadProps: UploadProps;
+  uploadModalVisibility: boolean;
+  displayUploadModal: () => void;
+  hideUploadModal: () => void;
+  handleUpload: () => void;
+};
+
+const LinkToAssetModal: React.FC<Props> = ({
```
Comment (KaWaite) [inline web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx:35]: LinkAssetModal

File: web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx [hunk 4]
```diff
@@ -0,0 +1,150 @@
...
+    {
+      title: t("Created By"),
+      dataIndex: "createdBy",
+      key: "createdBy",
+    },
+  ];
+
+  return (
+    <Modal
+      title={t("Link To Asset")}
```
Comment (KaWaite) [inline web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx:113]: Link Asset

File: web/src/components/molecules/Content/Form/index.tsx [hunk 5]
```diff
@@ -68,6 +91,10 @@ const ContentForm: React.FC<Props> = ({
     }
   }, [form, model?.schema.fields, model?.schema.id, itemId, onItemCreate, onItemUpdate]);
 
+  const handleLink = (_asset: Asset) => {
```
Comment (KaWaite) [inline web/src/components/molecules/Content/Form/index.tsx:94]: Does this mean this PR is just the UI?

File: web/src/components/molecules/Content/Form/index.tsx [hunk 6]
```diff
@@ -68,6 +91,10 @@ const ContentForm: React.FC<Props> = ({
...
+    // TODO: implement link asset with content
```
Comment (KaWaite) [inline web/src/components/molecules/Content/Form/index.tsx:95]: Does this mean this PR is just the UI?

File: web/src/components/molecules/Asset/UploadModal/uploadModal.tsx [hunk 7]
```diff
@@ -20,6 +22,8 @@ type Props = {
 };
 
 const UploadModal: React.FC<Props> = ({
+  uploadButtonText,
+  cancelButtonText,
```
Comment (KaWaite) [inline web/src/components/molecules/Asset/UploadModal/uploadModal.tsx:26]: Why are these needed? Couldn't find where they are being used.

File: web/src/components/molecules/Asset/UploadAsset/index.tsx [hunk 8]
```diff
@@ -6,6 +6,7 @@ import { useT } from "@reearth-cms/i18n";
 import UploadModal from "../UploadModal/uploadModal";
 
 type Props = {
+  text?: string;
```
Comment (KaWaite) [inline web/src/components/molecules/Asset/UploadAsset/index.tsx:9]: Why is this needed?

File: web/src/components/molecules/Common/Form/AssetItem/index.tsx (hunk 2 above)
Comment (nourbalaha) [reply web/src/components/molecules/Common/Form/AssetItem/index.tsx]: Done!

File: web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx (hunk 3 above)
Comment (nourbalaha) [reply web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx]: Done!

File: web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx (hunk 4 above)
Comment (nourbalaha) [reply web/src/components/molecules/Common/LinkToAssetModal/linkToAssetModal.tsx]: Done!

File: web/src/components/molecules/Content/Form/index.tsx (hunk 6 above)
Comment (nourbalaha) [reply web/src/components/molecules/Content/Form/index.tsx:95]: In this PR I connected Asset with Content since I developed Asset in the first place. and it seemed a bit confusing for Basel to Connect it.
So me and Basel decided that I will connect Asset with Content and Basel will handle the "Link Asset" part.

File: web/src/components/molecules/Asset/UploadModal/uploadModal.tsx (hunk 7 above)
Comment (nourbalaha) [reply web/src/components/molecules/Asset/UploadModal/uploadModal.tsx]: I removed unused arguments.

File: web/src/components/molecules/Asset/UploadAsset/index.tsx (hunk 8 above)
Comment (nourbalaha) [reply web/src/components/molecules/Asset/UploadAsset/index.tsx]: I removed unused arguments.

File: web/src/components/organisms/Project/Content/ContentDetails/index.tsx (hunk 1 above)
Comment (nourbalaha) [reply web/src/components/organisms/Project/Content/ContentDetails/index.tsx:33]: Please correct me if I am wrong, 
The order of the arguments(variables, methods) should be by:
1- Variables then methods.
2- Grouped by relativity to on another.
3- Alphabetically.
right?

Comment (nourbalaha) [reply web/src/components/organisms/Project/Content/ContentDetails/index.tsx:33]: In this case the order should be like this:
```js
  const {
    assetList,
    fileList,
    loading,
    uploading,
    uploadModalVisibility,
    createAssets,
    handleSearchTerm,
    handleAssetsReload,
    setFileList,
    setUploading,
    setUploadModalVisibility,
```

File: web/src/components/molecules/Content/Form/index.tsx (hunk 6 above)
Comment (nourbalaha) [reply web/src/components/molecules/Content/Form/index.tsx:95]: In this PR I connected `asset` with `content` since I developed asset in the first place and it seemed a bit confusing to Basel to connect it.
So, me and Basel have decided that I connect `asset` with `content` and he will handle the `link` part.

File: web/src/components/organisms/Project/Content/ContentDetails/index.tsx (hunk 1 above)
Comment (KaWaite) [reply web/src/components/organisms/Project/Content/ContentDetails/index.tsx:33]: Ah you don't need to worry about alphabetical order! But the other points are what I hope we can aim for as much as possible! 

Also, keeping the return order from the hooks. For example, if the hooks.ts file is:
```
return {
    assetList,
    fileList,
    loading,
    uploading,
    uploadModalVisibility,
}
```
then we just copy paste that order where its being used:
const {
    assetList,
    fileList,
    loading,
    uploading,
    uploadModalVisibility,
} = useHooks();

Comment (KaWaite) [reply web/src/components/organisms/Project/Content/ContentDetails/index.tsx:33]: If this seems like too much or annoying or anyways you just are against this approach, please fight me on it :P But for now, this is what I hope for.

File: web/src/components/molecules/Asset/UploadModal/uploadModal.tsx (hunk 7 above)
Comment (KaWaite) [reply web/src/components/molecules/Asset/UploadModal/uploadModal.tsx]: Instead of passing the `uploadButtonText` string, can you pass `alsoLink` (boolean)? 
I feel like passing uploadButtonText works, but its not really specific enough I feel. If the UploadModal component was more generic, being able to change the button text might make sense. But this is pretty specifically for uploading, or uploading and linking, assets.

Comment (nourbalaha) [reply web/src/components/molecules/Asset/UploadModal/uploadModal.tsx]: Done!

File: web/src/components/organisms/Project/Content/ContentDetails/index.tsx (hunk 1 above)
Comment (nourbalaha) [reply web/src/components/organisms/Project/Content/ContentDetails/index.tsx]: It is not too much at all. it is cleaner this way.
No need to fight you on it ;)
Done

**Review by KaWaite — APPROVED** [review]
